Write up everything on close form probs (#35)
11michalis11/AmbulanceDecisionGame: tex/main/MarkovChain/closed_form_state_probs/main.tex

\subsection{Graph Theoretical Approach}

> This section aims to investigate the connection between Markov chains and graph theory by exploring alternative ways of calculating the state probabilities and considering the problem from different perspectives.

Author: I decided to have one sentence per line. Do you think I should force break at 80 characters instead? Or was this comment just for removing the word "steady" in line 2?

> This section aims to investigate the connection between Markov chains and graph theory by exploring alternative ways of calculating the state probabilities and considering the problem from different perspectives.

Reviewer: I *added* the word steady and also broke the lines out of habit. It makes the diffs more precise.

\subsubsection{Parameters}

> The inputs of the formula are the number of servers \(C\), the threshold \(T\), the system capacity \(N\) and the parking capacity \(M\). 

Author: Agreed. I have this in my to-do list. I have to reword the entire document (not just this section) and simply didn't bother changing it in this branch. Was planning to crate a new branch to fix this for the entire document.

> The inputs of the formula are the number of servers \(C\), the threshold \(T\), the system capacity \(N\) and the parking capacity \(M\). 

Reviewer: Cool. In which case add a `% TODO` comment for now.

> Additional parameters of the model are the ambulance arrival rate, the others' arrival rate and the service rate, but for the purpose of this section these will remain unknown (\(\lambda^A, \lambda^o, \mu\)). 

Author: Same as comment above.

> Additional parameters of the model are the ambulance arrival rate, the others' arrival rate and the service rate, but for the purpose of this section these will remain unknown (\(\lambda^A, \lambda^o, \mu\)). 

Reviewer: Same reply.

    \item \textbf{System capacity (\(N\)):} Is the upper bound of \(v\) for all states \((u,v)\).
    \item \textbf{Parking capacity (\(M\)):} Is the upper bound of \(u\) for all states \((u,v)\) such that \(u \geq T\).
\end{itemize}

\subsubsection{Example figure of Markov Model}

\begin{figure}[h]
    \centering
    \input{MarkovChain/closed_form_state_probs/example_model_1352/main.tex}
    \caption{\(C=1, T=3, N=5, M=2\)}
    \label{fig:Markov_1352_example_for_closed_form}
\end{figure}

In figure \ref{fig:Markov_1352_example_for_closed_form} an example of such Markov model is shown where \(C=1\) meaning the only coefficient in front of any \(\mu\) is going to be \(1\), \(T=3\) which means that the \textit{left arm} of the model has a length of \(3\), \(N=5\) that indicates that the right-most states \((u,v)\) are of the form \((u,5)\) and \(M=2\) that equivalently shows that the bottom states are of the form \((2,v)\).

\subsubsection{Transforming the Markov chain model into graph theoretic model}

An additional approach that one may consider to get the state probabilities is the graph theoretical approach for state probabilities.

> Thus, it can be assumed that a Markov chain model \(M\) can be translated as a weighted directed graph \(G_M\) where every edge has a weight that corresponds to the rate of the edges of the Markov chain. 

Reviewer: ```suggestion
Thus, it can be assumed that a Markov chain model \(M\) can be translated as a weighted 
directed graph \(G_M\) where every edge has a weight that corresponds to the 
rate between the states of a given edge.
```

For example, consider the graph shown in figure \ref{fig:example_spanning_tree}. The graph points out a spanning tree that is rooted at vertex 3.

\begin{figure}[h]
    \centering
    \begin{tikzpicture}
        \node[state](u1){1};
        \node[state, right=of u1](u2){2};
        \node[state, right=of u2](u3){3};
        \node[state, right=of u3](u4){4};
        \node[state, below=of u2](u5){5};
        \node[state, below=of u3](u6){6};
        \node[state, below=of u4](u7){7};
        \draw[->, thick] (u1) -- (u2);
        \draw[->, thick] (u2) -- (u3);
        \draw[->, thick] (u4) -- (u3);
        \draw[->, thick] (u5) -- (u2);
        \draw[->, thick] (u6) -- (u5);
        \draw[->, thick] (u7) -- (u6);
    \end{tikzpicture}
    \caption{Spanning tree of a graph rooted at vertex 3}
    \label{fig:example_spanning_tree}
\end{figure}

Additionally, let us denote the set of all spanning trees of \(G\) as \(T(G)\) and the subset of \(T(G)\) that includes only the spanning trees that are rooted at vertex \(v\) as \(T_v(G)\). 

> The weight of a spanning tree \(t\) can be defined as the product of the weights of the edges it contains: 
> \[w(t)=\prod_{e \in t} w(e)\]

Reviewer: ```suggestion
The weight of a spanning tree \(t\) can be defined as the product of the weights of the edges it contains: 

\[w(t)=\prod_{e \in t} w(e)\]
```

\textbf{Theorem: Markov chain tree theorem} \cite{markov-chain-tree-theorem} \newline
\textit{Let M be an irreducible Markov chain on n states with stationary distribution \(\pi_1, \pi_2, \dots, \pi_n\). 
Let \(G_M\) be the directed graph associated with \(M\). 
Then the probability of being at state \(u\) is given by:}

\begin{equation}\label{markov-chain-tree-theorem}
    \pi_i = \frac{\sum_{t \in T_i(G_M)} w(t)}{\sum_{t \in T(G_M)}w(t)}
\end{equation}

> Equation \ref{markov-chain-tree-theorem} states that the probability of being at state \(u\) can be found by dividing the sum of the weights of all spanning trees rooted at \(u\) by the sum of the weights of all spanning trees of the graph. 
> Let us ignore the denominator of that fraction for now and focus only on the numerator denoted as \(\tilde{\pi}_i=\sum_{t \in T_i(G_M)} w(t)\)

Reviewer: ```suggestion
Equation \ref{markov-chain-tree-theorem} states that the probability of being at state \(u\) 
can be found by dividing the sum of the weights of all trees in \(T_v(G)\)
by the sum of the weights of all tress in \(T(G)\). 
Let us ignore the denominator of that fraction for now and focus only on the numerator denoted as \(\tilde{\pi}_i=\sum_{t \in T_i(G_M)} w(t)\)
```

 

\newpage
\subsubsection{Spanning Trees rooted at \((0,0)\)}

Let us now consider some examples of spanning trees that are rooted at \((0,0)\). 

> For each of the following examples the complete model is shown, then all possible spanning trees rooted at \((0,0)\) along with the weight associated with each spanning tree and finally the value of \(\tilde{\pi}_{(0,0)}\) which is the sum of all the weights of the spanning trees.

Reviewer: ```suggestion
For each of the following examples the complete graph \(G\) is shown, then all possible trees of \(T_{(0,0)}(G)\) along with the weight associated with each spanning tree.
As well as this, the sum of all the weights of the spanning trees denoted by \(\tilde{\pi}_{(0,0)}\) is also included.
```

    \begin{equation*}
        (\lambda^A)^2 \mu^4
    \end{equation*}
    \vspace*{\fill}
    \columnbreak
    \begin{figure}[H]
        \centering
        \scalebox{0.6}{\input{MarkovChain/closed_form_state_probs/spanning_trees_1122/main_1.tex}}
    \end{figure}
    \vspace*{\fill}
    \columnbreak
    \vspace*{0.3cm}
    \begin{equation*}
        \lambda^A \mu^5
    \end{equation*}
\end{multicols}

\begin{multicols}{4}
    \begin{figure}[H]
        \centering
        \scalebox{0.6}{\input{MarkovChain/closed_form_state_probs/spanning_trees_1122/main_2.tex}}
    \end{figure}
    \vspace*{\fill}
    \columnbreak
    \vspace*{0.3cm}
    \begin{equation*}
        \lambda^A \mu^5
    \end{equation*}
    \vspace*{\fill}
    \columnbreak
    \begin{figure}[H]
        \centering
        \scalebox{0.6}{\input{MarkovChain/closed_form_state_probs/spanning_trees_1122/main_3.tex}}
    \end{figure}
    \vspace*{\fill}
    \columnbreak
    \vspace*{0.3cm}
    \begin{equation*}
        \mu^6
    \end{equation*}
\end{multicols}

\begin{equation*}
    \tilde{\pi}_{(0,0)} = (\lambda^A)^2 \mu^4 + 2 \lambda^A \mu^5 + \mu^6
\end{equation*}

\newpage
\subsubsection{Conjecture of adding rows}

> Let us consider three Markov models with the same number of servers \(C=1\), the same threshold \(T=1\), the same system capacity \(N=2\) but different parking capacity \(M=1\), \(M=2\) and \(M=3\).

Reviewer: ```suggestion
Let us consider three Markov models with the same number of servers \(C=1\), the same threshold \(T=1\), 
the same system capacity \(N=2\) but \(M\in\{1, 2, 3\}\).
```

\begin{multicols}{3}
    \begin{figure}[H]
        \centering
        \scalebox{0.8}{\input{MarkovChain/closed_form_state_probs/example_model_1121/main.tex}}
        \caption{\(M=1\)}
    \end{figure}
    \columnbreak
    \begin{figure}[H]
        \centering
        \scalebox{0.8}{\input{MarkovChain/closed_form_state_probs/example_model_1122/main.tex}}
        \caption{\(M=2\)}
    \end{figure}
    \begin{figure}[H]
        \centering
        \scalebox{0.8}{\input{MarkovChain/closed_form_state_probs/example_model_1123/main.tex}}
        \caption{\(M=3\)}
    \end{figure}
\end{multicols}

> By increasing the parking capacity of the model it can be easily observed that the number of spanning trees rooted at \((0,0)\) increases as well since more combinations of paths can be generated using the new edges and vertices. The corresponding values of \(\tilde{\pi}_{(0,0)}\) of the three models are:
> 

Reviewer: ```suggestion
By increasing the parking capacity of the system it can be observed that \(|T_{(0,0)}(G)|\) increases as well since 
more combinations of paths can be generated using the new edges and vertices. 
The corresponding values of \(\tilde{\pi}_{(0,0)}\) of the three systems are:

```

\begin{align}
    M = 1: \tilde{\pi}_{(0,0)} &= \mu^4 + \mu^3 \lambda_A = \mu^3 (\mu + \lambda^A) \label{eq:rows-conjecture-1}\\
    M = 2: \tilde{\pi}_{(0,0)} &= \mu^6 + 2\mu^5 \lambda_A + \mu^4 (\lambda^A)^2 = \mu^4(\mu^2 + 2\mu \lambda_A + (\lambda^A)^2) = \mu^4 (\mu + \lambda^A) ^ 2 \label{eq:rows-conjecture-2}\\
    M = 3: \tilde{\pi}_{(0,0)} &= \mu^8 + 3 \mu^7 \lambda^A + 3 \mu^6 (\lambda^A)^2 + \mu^5(\lambda^A)^3 \nonumber \\
    &= \mu^5 (\mu^3 + 3 \mu ^2 \lambda^A + 3 \mu (\lambda^A)^2 + (\lambda^A)^3) \nonumber \\
    &= \mu^5 (\mu + \lambda^A) ^ 3 \label{eq:rows-conjecture-3}
\end{align}

It can be observed from equations (\ref{eq:rows-conjecture-1}),( \ref{eq:rows-conjecture-2}) and (\ref{eq:rows-conjecture-3}), that there is a noticeable relationship between them. 
Thus, a generalised formula for the value of \(\tilde{\pi}_{(0,0)}\) when \(C=1, T=1 \text{ and } N=1\) is given by:

\begin{equation}\label{eq:rows-conjecture-general}
    \tilde{\pi}_{(0,0)} = \mu^{(N+M)} (\mu + \lambda^A)^M
\end{equation}

> 

Reviewer: This is too small a set of experiments from which to make that claim.

So we just need to more carefully word this. Something like:

```
Note that in (\ref{eq:rows-conjecture-1}),( \ref{eq:rows-conjecture-2}) and (\ref{eq:rows-conjecture-3}), the following equation holds:

\begin{equation}\label{eq:rows-conjecture-general}
      \tilde{\pi}_{(0,0)} = \mu^{(N+M)} (\mu + \lambda^A)^M
  \end{equation}

This has relationship has been verified experimentally for ... and the data set is archived at ... % TODO
```

And we can talk about throwing some code together to just run for a while and collect data for that.

It is important to note here that the above property holds when the system capacity is greater than one as well (\(N \geq 1\)). 

> For instance let us consider a Markov model with \(C\) number of servers, a threshold of \(T\), a system capacity of \(N\) and a parking capacity of \(M\). 

Reviewer: ```suggestion
For instance let us consider a system with \(C\) servers, a threshold of \(T\), a system capacity of \(N\) and a parking capacity of \(M\). 
```

> For instance let us consider a Markov model with \(C\) number of servers, a threshold of \(T\), a system capacity of \(N\) and a parking capacity of \(M\). 

Reviewer: This doesn't really follow from the previous sentence or lead to the next one.

The equivalent values of \(\tilde{\pi}_{(0,0)}\) can be expressed in terms of an unknown function \(k(C,T,N)\) as:

\begin{equation}
    \tilde{\pi}_{(0,0)} = \mu^{(N+M)} (k(C,T,N))^M
\end{equation}

> 

Reviewer: We need a little paragraph here transitioning to the next section.

> \subsubsection{Matrix-tree theorem for directed graphs (Kirchhoff's theorem) \cite{matrix-tree-theorem}: }

Reviewer: This comes out of nowhere.

Author: This is the theorem that I didn't use. It just helped get some insight. I basically "left" it there to so it could be used at some point when we come back to this

\textit{The number of directed spanning trees rooted at a state \(i\) can be found by calculating the determinant of the Laplacian matrix \(Q\) of the directed graph and removing row \(i\) and column \(i\).}

> \subsubsection{Conjecture of adding columns (combinatorics attempt for closed form formula)}

Reviewer: ```suggestion
\subsubsection{The effect of increasing \(N\)}
```

(Is that right or is it `\(M\)`?)

Reviewer: Also: combinatorics and combinations is not the same thing.

Everything here about enumerating the trees is combinatorial: combinations and permutations are both tools used in combinatorics.

Author: Oh okay, makes sense. Thanks for clearing that out.

> Having equation (\ref{eq:rows-conjecture-general}) it only remains to get the function \(k(C,T,N)\) that is independent of the value of the parking capacity. 
> Thus, in this section only models with a parking capacity of \(M=1\) will be considered and will be referred to as the base cases. 

Reviewer: ```suggestion
In this section we will consider a parking capacity of \(M\) and 
see the effect of modifying other parameters. On \(k(C, T, N)\).
```

For the starting coefficients of the model the function \(G(p_1,p_2,p_3)\) gives the values of the coefficients and is defined as:

\begin{equation} \label{eq:coefficient-function}
    G(p_1,p_2,p_3) = 
    \begin{cases}
        0 & \text{if } p_1 = 0 \text{ and } p_2 > 0 \vspace{0.1cm} \\
        1 & \text{if } p_1, p_2 = 0 \text{ and } p_3 > 0 \vspace{0.2cm} \\
        \binom{p_1 + p_3}{p_3} & \text{if } p_2 = 0 \text{ and } p_1 > 0 \vspace{0.2cm} \\
        \binom{p_1 + p_2 - 1}{p_2} & \text{if } p_3 = 0 \text{ and } p_1, p_2 > 0 \vspace{0.2cm} \\
        p_3 + 1 & \text{if } p_1 = 1 \vspace{0.2cm} \\
        \binom{p_1 + p_3 + 1}{p_1} + p_3 \binom{p_1 + p_3}{p_3+1} - \binom{p_1 + p_3}{p_3} & \text{if } p_2 = 1 \text{ and } p_1 > 1 \vspace{0.2cm} \\
        \binom{p_1 + p_2 + 1}{p_1} - \binom{p_1 + p_2 - 1}{p_2-1} + \sum_{i=p_2}^{p_1 + p_2 - 2} i \binom{i-1}{p_2 - 1} & \text{if } p_3 = 1 \text{ and } p_1,p_2 > 1 \vspace{0.2cm} \\
        U_{p_1,p_2,p_3} & \text{otherwise}
    \end{cases}
\end{equation}

Note here that the final value \(U_{p_1,p_2,p_3}\) corresponds to coefficients that are unknown and are currently investigated. 

> In essence function \(G()\) takes as arguments a possible combination of numbers of \(\lambda^A, \lambda^o \text{ and } \mu \) for a given Markov model and outputs the coefficient of that term which in turn represents how many spanning trees exist in the graph with that specific combination. 

Reviewer: ```suggestion
The function \(G\) takes as arguments a possible combination of numbers of 
\(\lambda^A, \lambda^o \text{ and } \mu \) for a given system and outputs the coefficient
 of that term which in turn represents how many spanning trees exist in the graph
  with that specific combination. 
```

        \item (\ref{eq:00_rate_1141}) \( \Rightarrow 3(\lambda^A)^2 \mu\): \(G(2,0,1) = \binom{2+1}{1} = 3\)
        \item (\ref{eq:00_rate_1141}) \( \Rightarrow 3 \lambda^A \mu^2\): \(G(1,0,2) = \binom{1+2}{2} = 3\)
        \item (\ref{eq:00_rate_1151}) \( \Rightarrow 3 (\lambda^A)^3 \lambda^o\): \(G(3,1,0) = \binom{3+1-1}{1} = 3\)
        \item (\ref{eq:00_rate_1151}) \( \Rightarrow 3 (\lambda^A)^2 (\lambda^o)^2 \): \(G(2,2,0) = \binom{3}{2} = 3\)
        \item (\ref{eq:00_rate_1151}) \( \Rightarrow 6 (\lambda^A)^2 \mu ^ 2\): \(G(2,0,2) = \binom{2+2}{2} = 6\)
    \end{itemize}
\end{multicols}

\begin{itemize}
    \item (\ref{eq:00_rate_1151}) \( \Rightarrow 6 (\lambda^A)^2 \lambda^o \mu\): \(G(2,1,1) = \binom{2+1+1}{2} + 1\binom{2+1}{1+1} - \binom{2+1}{1} = 6 + 3 - 3 = 6\)
    \item \small{(e.g)} \( \Rightarrow (\lambda^A)^2 (\lambda^o)^2 \mu\): \(G(2,2,1) = \binom{2+2+1}{2} - \binom{2+2-1}{2-1} + \sum_{i=2}^{2+2-2} i\binom{i-1}{2-1} = 10 - 3 + (2 \times 1) = 9\)
\end{itemize}

\subsubsection{Unknown terms}

The terms that remain unknown are the terms where \(p_1, p_2, p_3 \geq 2\). 

> For small models these terms are very few but as the model grows the number of these terms increase exponentially. 

Reviewer: Replace this sentence with a table of the numbers. "very few" is imprecise.

Here are some of these values with the corresponding values of the \(G(p_1,p_2,p_3)\) function.

\begin{multicols}{2}
    \begin{itemize}
        \item \(G(2,2,2) = 18\) 
        \item \(G(3,2,2) = 60\)
        \item \(G(2,3,2) = 24\)
        \item \(G(2,2,3) = 30\)
        \item \(G(4,2,2) = 150\)
        \item \(G(3,3,2) = 100\)
        \item \(G(3,2,3) = 120\)
        \item \(G(2,4,2) = 30\)
        \item \(G(2,3,3) = 40\)
        \item \(G(2,2,4) = 45\)
    \end{itemize}
\end{multicols}

> \subsubsection{Transforming the spanning tree problem into a permutation problem}

Reviewer: ```suggestion
\subsubsection{LDR Arrays}
```

> Given the nature of the spanning tree problem and the properties that the specific Markov chain model holds, this problem can be further reduced to another problem that may be slightly easier to solve. 
> Consider the following Markov model and the spanning trees rooted at state \((0,0)\) that are associated with it. 

Reviewer: ```suggestion
In this section a new combinatorial object is defined: LDR arrays. It will be shown that there is a 
bijection between LDR arrays and the ... . LDR arrays will then be enumerated which in 
turn enumerates the ....
```

\subsection{Graph Theoretical Approach}

> % TODO: Generalise notation in entire section 

Reviewer: A big general `% TODO` is fine but also include specific ones at each of the lines I pointed out (and any others). It means that work (me reviewing and finding things) doesn't have to be duplicated.

This section aims to investigate the connection between Markov chains and graph theory by exploring alternative ways of calculating the state probabilities and considering the problem from different perspectives.
This section aims to investigate the connection between Markov chains and graph theory by exploring alternative ways of calculating the steady state probabilities and considering the problem from different perspectives.

\subsubsection{Parameters}

> The inputs of the formula are the number of servers \(C\), the threshold \(T\), the system capacity \(N\) and the parking capacity \(M\). 

Reviewer: You missed the first sentence in my suggestion.

Author: Oops. My bad

    \item the threshold \(T\), 
    \item the system capacity \(N\),
    \item the parking capacity \(M\). 
\end{itemize}
Additional parameters of the model are the ambulance arrival rate, the others' arrival rate and the service rate (\(\lambda^A, \lambda^o, \mu\)). 
More specifically, the way these parameters are translated into the model are:

\begin{itemize}

>     \item \textbf{Number of servers (\(C\)):} All edges in the Markov chain that correspond to a service rate have a weight of: 

Reviewer: I meant that you would simply point at your definition of the transition rates?

Author: Oh okay. So just reference the equation where I defined these transition rates?

Reviewer: I think so. I think that's sufficient no. What do you think?

Author: I am not sure. I wanted to explain it in terms of a graph theoretic perspective and I felt that I should use different notation to define stuff. But it does make sense to have it that way because it is essentially the same thing,

Reviewer: I understand. I'd suggesting solving both problems. I forget your exact notation but let's say 

- `S` is the state space of the Markov chain
- `q_{s_1, s_2}` is the transition rate between `s_1` and `s_2`.
- `w_{e}` is the weight for edge `e`.

Then I suggest we go for something like:

```
We define the weighted digraph \(G=(V, E)\) where \(V=S\) and \((v_1, v_2)\in E\) 
if and only if \(q_{v_1, v_2}>0\). Further more, the weights are given by:

\[
    w_((v_1, v_2)) = q_{v_1, v_}
\]

This way we're rigorously defining the graph using the correct vocabulary but also clearly showing the connection back to the chain?
```

Author: Ohhh. I like this. Yeah I will include something like this in.

\begin{figure}[h]
    \centering
    \scalebox{0.6}{\input{MarkovChain/closed_form_state_probs/example_model_1352/main.tex}}
    \caption{\(C=1, T=3, N=5, M=2\)}
    \label{fig:Markov_1352_example_for_closed_form}
\end{figure}

> In figure \ref{fig:Markov_1352_example_for_closed_form} an example of such Markov model is shown where \(C=1\), \(T=3\) which means that the \textit{left arm} of the model has a length of \(3\), \(N=5\) that indicates that the right-most states \((u,v)\) are of the form \((u,5)\) and \(M=2\) that equivalently shows that the bottom states are of the form \((2,v)\).

Reviewer: ```suggestion
In figure \ref{fig:Markov_1352_example_for_closed_form} an example of such a Markov model is shown where \(C=1\), \(T=3\) which means that the \textit{left arm} of the model has a length of \(3\), \(N=5\) that indicates that the right-most states \((u,v)\) are of the form \((u,5)\) and \(M=2\) that equivalently shows that the bottom states are of the form \((2,v)\).
```

    &= \mu^5 (\mu + \lambda^A) ^ 3 \label{eq:rows-conjecture-3}
\end{align}

Note that in equations (\ref{eq:rows-conjecture-1}),(\ref{eq:rows-conjecture-2}) and (\ref{eq:rows-conjecture-3}), the following equation holds: 

\begin{equation}\label{eq:rows-conjecture-general}
    \tilde{\pi}_{(0,0)} = \mu^{(N+M)} (\mu + \lambda^A)^M
\end{equation}

% TODO: Perform experiments to validate this
This relationship has been verified experimentally for ... and the data set is archived at ... 
A generalisation of equation \ref{eq:rows-conjecture-general}, where \(N \geq 1\), is given in terms of an unknown function \(k(C,T,N)\) as:

\begin{equation}
    \tilde{\pi}_{(0,0)} = \mu^{(N+M)} (k(C,T,N))^M
\end{equation}

> Thus, having investigated the effect of adding rows it remains to investigate the effect of increasing \(N\) and finding an expression for \(k(C,T,N)\).

Reviewer: ```suggestion
Thus, having investigated the effect of adding rows (increasing \(N\)) it remains 
to investigate the effect of adding columns (increasing \(N\)) and 
finding an expression for \(k(C,T,N)\).
```

\end{equation}

Note here that the final value \(U_{p_1,p_2,p_3}\) corresponds to coefficients that are unknown and are currently investigated. 

> The function \(R()\) takes as arguments a possible combination of numbers of \(\lambda^A, \lambda^o \text{ and } \mu \) for a given system and outputs the coefficient of that term which in turn represents how many spanning trees exist in the graph with that specific combination. 

Reviewer: You're missing things from my suggestion (smaller diffs would make this less likely to happen).

```suggestion
The function \(R\) takes as arguments a possible combination of numbers of \(\lambda^A, \lambda^o \text{ and } 
\mu \) for a given system and outputs the coefficient of that term which in turn represents how many spanning
 trees exist in the graph with that specific combination. 
```

Reviewer: Why the change from `G` to `R`?

Reviewer: I was thinking about this over breakfast and it's because we're using \(G\) to refer to the graph right? (Make sense :))

Author: Yeah was just about to reply. That's exactly why. G_M is the graph associated with the Markov chain

Reviewer: Cool. Makes sense (but yeah things like these, document them in the commit message body or a comment on the PR :)).

    \end{itemize}
\end{multicols}

\begin{itemize}
    \item (\ref{eq:00_rate_1151}) \( \Rightarrow 6 (\lambda^A)^2 \lambda^o \mu\): \(R(2,1,1) = \binom{2+1+1}{2} + 1\binom{2+1}{1+1} - \binom{2+1}{1} = 6 + 3 - 3 = 6\)
    \item \small{(e.g)} \( \Rightarrow (\lambda^A)^2 (\lambda^o)^2 \mu\): \(R(2,2,1) = \binom{2+2+1}{2} - \binom{2+2-1}{2-1} + \sum_{i=2}^{2+2-2} i\binom{i-1}{2-1} = 10 - 3 + (2 \times 1) = 9\)
\end{itemize}

\subsubsection{Unknown terms}

The terms that remain unknown are the terms where \(p_1, p_2, p_3 \geq 2\). 
Here are some of these values with the corresponding values of the \(R(p_1,p_2,p_3)\) function.

\begin{multicols}{2}
    \begin{itemize}
        \item \(R(2,2,2) = 18\) 
        \item \(R(3,2,2) = 60\)
        \item \(R(2,3,2) = 24\)
        \item \(R(2,2,3) = 30\)
        \item \(R(4,2,2) = 150\)
        \item \(R(3,3,2) = 100\)
        \item \(R(3,2,3) = 120\)
        \item \(R(2,4,2) = 30\)
        \item \(R(2,3,3) = 40\)
        \item \(R(2,2,4) = 45\)
    \end{itemize}
\end{multicols}

\subsubsection{DRL arrays}

In this section a new combinatorial object is defined: DRL arrays. 

> It will be shown that there is a bijection between DRL arrays and the spanning trees in \(G_M\).

Reviewer: I missed where we defined `\(G_M\)`?

Author: Line 48. First sentence of section 6.3.3

Reviewer: Yup perfect. Thank you.
